src/main/java/com/google/cloud/tools/shell/CommandLine.java

```diff
@@ -0,0 +1,80 @@
+/* Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.shell;
+
+import com.google.common.annotations.Beta;
+
+import java.util.ArrayList;
+
+/**
+ * Utilities for working with command line arguments.
+ */
+@Beta
+public class CommandLine {
+
+  /**
+   * Split a full command line into an array of individual arguments,
+   * similar to the shlex function in Python according to POSIX rules. 
+   * All input characters are preserved except for separating whitespace. 
+   * This function tokenizes the input string, but does not attempt to parse it.
+   * For example, it recognizes that a space escaped with a backslash does not
+   * start a new token, but it does not replace the "\ " with a single space. 
+   * 
+   * @param line the input line
+   * @return a non-null but possibly empty array of arguments
+   */
+  public static String[] split(String line) {    
+    char quote = '"';
```

Comment:
It needs to be initialized to something though:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project appengine-plugins-core: Compilation failure
[ERROR] /Users/elharo/appengine-plugins-core/src/main/java/com/google/cloud/tools/shell/CommandLine.java:[52,35] variable quote might not have been initialized
[ERROR] -> [Help 1]
[ERROR]